File: Home/app/src/main/java/com/faizzakiramadhan_19104075/myapplication/FoodLocationAdapter.java
package com.faizzakiramadhan_19104075.myapplication;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.flaviofaria.kenburnsview.KenBurnsView;
import com.squareup.picasso.Picasso;

import java.util.List;

public class FoodLocationAdapter extends RecyclerView.Adapter<FoodLocationAdapter.FoodLocationViewHolder> {

    private List<Food> foodlocation;

    public FoodLocationAdapter(List<Food> foodlocation) {
        this.foodlocation = foodlocation;
    }

    @NonNull
    @Override
    public FoodLocationViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new FoodLocationViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(
                        R.layout.card_item,
                        parent,
                        false
                )
        );
    }

    @Override
    public void onBindViewHolder(@NonNull FoodLocationViewHolder holder, int position) {

        holder.setLocationData(foodlocation.get(position));

    }

    @Override
    public int getItemCount() {
        return foodlocation.size();
    }

    static class FoodLocationViewHolder extends RecyclerView.ViewHolder{

        private KenBurnsView kbv_location;
        private TextView text_title, text_location, text_starRating;


        FoodLocationViewHolder(@NonNull View itemView) {
            super(itemView);
            kbv_location = itemView.findViewById(R.id.kbv_location);
            text_title = itemView.findViewById(R.id.text_title);
            text_location = itemView.findViewById(R.id.text_location);
            text_starRating = itemView.findViewById(R.id.text_starRating);
        }

        void setLocationData(Food food){
            Picasso.get().load(food.imageUrl).into(kbv_location);
            text_title.setText(food.title);
            text_location.setText(food.harga);
            text_starRating.setText(String.valueOf(food.starRating));
        }

    }
}

File: Home/app/src/main/java/com/faizzakiramadhan_19104075/myapplication/FoodLocationAdapter.kt
package com.faizzakiramadhan_19104075.myapplication

import com.faizzakiramadhan_19104075.myapplication.Food
import androidx.recyclerview.widget.RecyclerView
import com.faizzakiramadhan_19104075.myapplication.FoodLocationAdapter.FoodLocationViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.faizzakiramadhan_19104075.myapplication.R
import com.flaviofaria.kenburnsview.KenBurnsView
import android.widget.TextView
import com.squareup.picasso.Picasso
import java.lang.String

class FoodLocationAdapter(private val foodlocation: List<Food>) :
    RecyclerView.Adapter<FoodLocationViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FoodLocationViewHolder {
        return FoodLocationViewHolder(
            LayoutInflater.from(parent.context).inflate(
                R.layout.card_item,
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: FoodLocationViewHolder, position: Int) {
        holder.setLocationData(foodlocation[position])
    }

    override fun getItemCount(): Int {
        return foodlocation.size
    }

    class FoodLocationViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val kbv_location: KenBurnsView
        private val text_title: TextView
        private val text_location: TextView
        private val text_starRating: TextView
        fun setLocationData(food: Food) {
            Picasso.get().load(food.imageUrl).into(kbv_location)
            text_title.text = food.title
            text_location.text = food.harga
            text_starRating.text = String.valueOf(food.starRating)
        }

        init {
            kbv_location = itemView.findViewById(R.id.kbv_location)
            text_title = itemView.findViewById(R.id.text_title)
            text_location = itemView.findViewById(R.id.text_location)
            text_starRating = itemView.findViewById(R.id.text_starRating)
        }
    }
}